infra/modules/azure_event_hub/variables.tf

```
@@ -63,6 +63,15 @@ variable "subnet_pep_id" {
   description = "Id of the subnet which holds private endpoints"
 }
 
+variable "allowed_sources" {
+  type = list(object({
+    subnet_ids = list(string)
+    ips        = list(string)
+  }))
+  default     = []
+  description = "List of allowed sources (subnets and ip masks)"
+}
+
```

Do we actually need a list of objects with lists inside?
Maybe the object is sufficient
```suggestion
variable "allowed_sources" {
  type = object({
    subnet_ids = optional(list(string), [])
    ips        = optional(list(string), [])
  })
  default     = {}
  description = "List of allowed sources (subnets and ip masks)"
}

```

---

it depdends. if subnet and ips are related is better to use a list of objects, otherwise an object with two lists suffices

---

I think @christian-calabrese is right, making a list of objects can create confusion for those who configure the variable, better an object with just the two lists inside for subnets and ips